### PR TITLE
Small step on docs

### PR DESCRIPTION
**Pull request by nedbat**
**Description:** 

- Wrote more in the docs.
- Reorganized the docs.
  - Readme isn't duplicated in the docs
  - Autodoc is less automatic: we edit the api.rst file to include the public parts
- Added an APIView to the examples directory.
- Tweaked how the tested schema is compared.
- Fixed some dependencies.

### CODE REVIEW

**Comment by nedbat** [comment]
@kdmccormick I've made a number of changes.  There's more to be written, and I haven't even looked at the docstrings yet.  But better to keep the feedback cycles short.

File: docs/writing.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
+.. _writing:
+
+Writing API documentation
+=========================
+
+Documentation for API endpoints starts in the Python files, using docstrings
+and decorators.
+
+
+For an APIView
```
Comment (kdmccormick) [inline docs/writing.rst]: Unfortunately the usage of `schema` vs. `schema_for` is not as simple as APIView vs. ViewSet.

APIViews expose the HTTP verbs `get`, `post`, `put`, `patch`, and `delete` as methods, whereas ViewSets expose "abstract" HTTP verbs `list`, `retrieve`, `create`, `update`, `partial_update`, and `destroy` as methods. Any of those methods, if defined by the library user, can be documented directly with `@schema`, and drf-yasg will figure out what's going on.

On either of those base classes, though, you can avoid overriding the HTTP-verb-like methods at all by instead implementing functions like `get_object`, `get_queryset`, `get_serializer`, `perform_create`, etc. This is where `@schema_for` becomes necessary.

I recommend splitting the two subsections along the lines of _Documenting methods directly with @ schema_ and _Documenting methods indirectly with @schema_for_

File: docs/api.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
+edx_api_doc_tools package
+============================
```
Comment (kdmccormick) [inline docs/api.rst]: Note to self: I need to clean up some of my docstrings so that they render better on ReadTheDocs.

File: docs/api.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+edx_api_doc_tools.apps module
+-----------------------------
+
+.. automodule:: edx_api_doc_tools.apps
+   :members:
+
+edx_api_doc_tools.conf_utils module
```
Comment (kdmccormick) [inline docs/api.rst]: Can we expose docs for  `edx_api_doc_tools.__init__` instead?

I imagined the `data`/`conf_utils`/`view_utils` separation as being an internal code-organization thing, not an interface.

File: docs/writing.rst (hunk 1 above)
Comment (kdmccormick) [reply docs/writing.rst]: If this contradicts what I've said before, my bad; the more I try to understand DRF the less I feel like I actually understand it.

Comment (nedbat) [reply docs/writing.rst]: RIGHT! This would go better if I devoted more of my brain to it... I'll make those changes.

File: docs/api.rst (hunk 2 above)
Comment (nedbat) [reply docs/api.rst]: I've tweaked some of the docstrings.

File: docs/api.rst (hunk 3 above)
Comment (nedbat) [reply docs/api.rst]: Done.

**Comment by nedbat** [comment]
@kdmccormick I think this might be mergeable....

**Pull request merged**